### PR TITLE
Support MFA device for cross account role switching

### PR DESCRIPTION
**Pull request by pdecat**
Support MFA device for cross account role switching, reusing STS credentials between API calls as MFA Token cannot be used more than once.

This introduces two new optional command line arguments:

      -M MFA_SERIAL_NUMBER, --mfa-serial-number MFA_SERIAL_NUMBER
                        MFA Serial Number to use when assuming a role via STS
      -T MFA_TOKEN, --mfa-token MFA_TOKEN
                        MFA Token to use when assuming a role via STS

### CODE REVIEW

**Comment by jantman** [comment]
@pdecat Please accept my most sincere apologies; it seems like I never got the notification for this.

The only Travis failure was a minor formatting issue in one of the doc strings, and coverage is complete. I'll look this over tonight or tomorrow and get it merged. I'm hoping to have a patch-level release out within the next few days.

File: awslimitchecker/checker.py [hunk 1]
```diff
@@ -83,6 +83,12 @@ def __init__(self, warning_threshold=80, critical_threshold=99,
           com/IAM/latest/UserGuide/id_roles_create_for-user_externalid.html>`_
           string to use when assuming a role via STS.
         :type external_id: str
+        :param mfa_serial_number: (optional) the `MFA Serial Number` string to_
+        use when assuming a role via STS.
```
Comment (jantman) [inline awslimitchecker/checker.py]: The continuation line here and on line 89 needs to be indented two spaces. That's why the [Travis doc test](https://travis-ci.org/jantman/awslimitchecker/jobs/95972816#L278) is failing.

File: awslimitchecker/checker.py [hunk 2]
```diff
@@ -83,6 +83,12 @@ def __init__(self, warning_threshold=80, critical_threshold=99,
...
+        :type mfa_serial_number: str
+        :param mfa_token: (optional) the `MFA Token` string to use when_
```
Comment (jantman) [inline awslimitchecker/checker.py]: The continuation line here and on line 87 needs to be indented two spaces. That's why the [Travis doc test](https://travis-ci.org/jantman/awslimitchecker/jobs/95972816#L278) is failing.

**Comment by jantman** [comment]
It looks like there are a few more places that have indentation issues causing the docs build to fail. Also, there should probably be something added to the documentation about this - I'd really like to see it in the "Credentials" section of the "Getting Started" page.

Since these are such trivial changes, I'll leave it up to you whether you want to fix them, or just want me to pull this in to my own branch and clean it up a bit.

I've never used MFA with STS before, so I'm going to need to just take your word that this works...

**Comment by jantman** [comment]
Ok, thanks! I'll merge this and have a release out tonight or early tomorrow morning (GMT-5)

**Comment by pdecat** [comment]
You're welcome.

I'll try to find some time to add this to the Credentials section.

**Comment by jantman** [comment]
@pdecat don't worry about it, I'll add the docs after merging.

Thanks so much for contributing this!

**Comment by jantman** [comment]
@pdecat this has been released as 0.2.3 and is now live on pypi. Thanks again so much for the contribution!

**Pull request merged**